schema/std-unpriv-extensions.asn

```diff
@@ -0,0 +1,715 @@
+BEGIN
+    EXPORTS Zve64d;
+
+    Zve64d ::= SEQUENCE {
+        version Version OPTIONAL,
+        ...
+    }
+
+END -- Zve64d-Extension
+
+Zve64f-Extension -- Zve64f: Embedded Vector Computation, FP32, 64 Minimum VLEN.  Implies F or Zfinx.
+DEFINITIONS
+    AUTOMATIC TAGS ::=
+BEGIN
+    EXPORTS Zve64f;
+
+    Zve64f ::= SEQUENCE {
+        version Version OPTIONAL,
+        ...
+    }
+
+END -- Zve64f-Extension
+
+Zve64x-Extension -- Zve64x: Embedded Vector Computation, no FP, 64 Minimum VLEN
+DEFINITIONS
+    AUTOMATIC TAGS ::=
+BEGIN
+    EXPORTS Zve64x;
+
+    Zve64x ::= SEQUENCE {
+        version Version OPTIONAL,
+        ...
+    }
+
+END -- Zve64x-Extension
+
+Zk-Extension -- Zk: Standard Scalar Cryptography.  Implies Zkn, Zkr, and Zkt
+DEFINITIONS
+    AUTOMATIC TAGS ::=
+BEGIN
+    EXPORTS Zk;
+
+    Zk ::= SEQUENCE {
+        version Version OPTIONAL,
+        ...
+    }
+
+END -- Zk-Extension
+
+Zkn-Extension -- Zkn: NIST Cryptography, Implies Zbkb, Zbkc, Zbkx, Zkne, Zknd, and Zknh
+DEFINITIONS
+    AUTOMATIC TAGS ::=
+BEGIN
+    EXPORTS Zkn;
+
+    Zkn ::= SEQUENCE {
+        version Version OPTIONAL,
+        ...
+    }
+
+END -- Zkn-Extension
+
+Zknd-Extension -- Zknd: NIST Suite: AES Decryption
+DEFINITIONS
+    AUTOMATIC TAGS ::=
+BEGIN
+    EXPORTS Zknd;
+
+    Zknd ::= SEQUENCE {
+        version Version OPTIONAL,
+        ...
+    }
+
+END -- Zknd-Extension
+
+Zkne-Extension -- Zkne: NIST Suite: AES Encryption
+DEFINITIONS
+    AUTOMATIC TAGS ::=
+BEGIN
+    EXPORTS Zkne;
+
+    Zkne ::= SEQUENCE {
+        version Version OPTIONAL,
+        ...
+    }
+
+END -- Zkne-Extension
+
+Zknh-Extension -- Zknh: NIST Suite: SHA2 Hashing
+DEFINITIONS
+    AUTOMATIC TAGS ::=
+BEGIN
+    EXPORTS Zknh;
+
+    Zknh ::= SEQUENCE {
+        version Version OPTIONAL,
+        ...
+    }
+
+END -- Zknh-Extension
+
+Zkr-Extension -- Zkr: Entropy Source Extension
+DEFINITIONS
+    AUTOMATIC TAGS ::=
+BEGIN
+    EXPORTS Zkr;
+
+    Zkr ::= SEQUENCE {
+        version Version OPTIONAL,
+        ...
+    }
+
+END -- Zkr-Extension
+
+Zks-Extension -- Zks: ShangMi Algorithm Cryptography.  Implies Zbkb, Zbkc, Zbkx, Zksed, and Zksh.
+DEFINITIONS
+    AUTOMATIC TAGS ::=
+BEGIN
+    EXPORTS Zks;
+
+    Zks ::= SEQUENCE {
+        version Version OPTIONAL,
+        ...
+    }
+
+END -- Zks-Extension
+
+Zksed-Extension -- Zksed: ShangMi Suite: SM4 Block Cipher
+DEFINITIONS
+    AUTOMATIC TAGS ::=
+BEGIN
+    EXPORTS Zksed;
+
+    Zksed ::= SEQUENCE {
+        version Version OPTIONAL,
+        ...
+    }
+
+END -- Zksed-Extension
+
+Zksh-Extension -- Zksh: ShangMi Suite: SM3 Hashing
+DEFINITIONS
+    AUTOMATIC TAGS ::=
+BEGIN
+    EXPORTS Zksh;
+
+    Zksh ::= SEQUENCE {
+        version Version OPTIONAL,
+        ...
+    }
+
+END -- Zksh-Extension
+
+Zkt-Extension -- Zkt: Data Independent Execution Latency
+DEFINITIONS
+    AUTOMATIC TAGS ::=
+BEGIN
+    EXPORTS Zkt;
+
+    Zkt ::= SEQUENCE {
+        version Version OPTIONAL,
+        ...
```

Comment:
Just want to confirm that every single one of these extensions looks exactly like this one, and I didn't miss an odd one out somewhere.

That is a good starting point for future work where individual extensions might need to add additional optional configuration data.

Reply:
yes